src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -320,24 +309,25 @@
       (let [job-pool (mk-pool)
             gc-interval-millis (to-millis gc-interval)
             gc-task #(interspaced gc-interval-millis % job-pool)
-            db-maintenance-tasks [(when (pos? (to-seconds node-ttl))
-                                    #(auto-expire-nodes!
-                                      node-ttl % mq-connection))
-                                  (when (pos? (to-seconds node-purge-ttl))
-                                    (partial purge-nodes! node-purge-ttl))
-                                  (when (pos? (to-seconds report-ttl))
-                                    (partial sweep-reports! report-ttl))
-                                  ;; Order is important here to ensure
-                                  ;; anything referencing an env or resource
-                                  ;; param is purged first
-                                  garbage-collect!]]
+            seconds-pos? (comp pos? to-seconds)
+            db-maintenance-tasks #(do 
```

Comment:
For such a large anonymous fn, I prefer (fn []...) I have to scan quite a few lines here to figure out if there is an argument or not to the fn